application/constants/templates.py

```diff
@@ -43,3 +49,51 @@ class HookTypes(StrEnum):
     Template application hook types.
     """
     kubernetes_job = 'kubernetes_job'
+
+
+class http(StrEnum):
+    """
+    Template HTTP component types.
+    """
+    url = 'url'
+    method = 'method'
+    headers = 'headers'
+    parameters = 'parameters'
+
+class httpMethods(StrEnum):
+    """
+    Template HTTP component methods.
+    """
+    get = 'GET'
+    post = 'POST'
+    put = 'PUT'
+    patch = 'PATCH'
+    delete = 'DELETE'
+
+
+class httpDeploy(BaseModel):
+    """
+    Deploy schema for HTTP type components.
+    """
+    url: constr(min_length=1, strip_whitespace=True) = Field(description='URL for HTTP deployment', example='http://example.com')
+    method: Union[httpMethods, constr(min_length=1, strip_whitespace=True)] = Field(default=httpMethods.get, description='HTTP method for deployment', example=httpMethods.post)
+    headers: Optional[dict] = Field(description='Headers for HTTP deployment', example={'Content-Type': 'application/json'})
```

Comment:
We have Python version that allows to use modern more simpler type notation. So ` headers: Optional[dict]` can be replaced by ` headers: dict | None` or I assume we can go further and define it like: ` headers: dict[str, str] | None`.

Reply:
Also I think it should be `dict[str, str]`. I mean not optional but empty dict is acceptable value. But this is optional comment.